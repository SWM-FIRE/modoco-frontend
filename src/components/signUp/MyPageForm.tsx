import styled from 'styled-components';
import Avatar from './Avatar';
import Nickname from './Nickname';
import useChangeProfile from '../../hooks/useChangeProfile';

export default function MyPageForm() {
  const { inputs, onChange, onSubmit, onChangeAvatar, isDisable } =
    useChangeProfile();
  const { avatar, nickname } = inputs;
  return (
    <Container>
      <Title>프로필 수정</Title>
      <Form onSubmit={onSubmit}>
        <Avatar avatar={avatar} onChangeAvatar={onChangeAvatar} />
        <Nickname nickname={nickname} onChange={onChange} />
        <Submit disabled={isDisable()}>수정 완료</Submit>
      </Form>
    </Container>
  );
}

const Container = styled.div`
  background-color: #18181b;
  display: flex;
  flex-direction: column;
  width: 61rem;
  align-items: flex-start;
  margin: 0 auto;
  padding: 7.8rem 0 14.4rem 0;
  font-family: IBMPlexSansKRRegular;
`;

const Title = styled.h1`
  font-size: 4rem;
  color: #f9fafb;
`;

const Form = styled.form`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  width: 100%;

  input:focus {
    outline: none;
  }
`;

const Submit = styled.button`
  height: 5.5rem;
  background-color: #f3f4f6;
  color: #111827;
  width: 100%;
  font-size: 1.5rem;
  border-radius: 1rem;
  font-family: IBMPlexSansKRRegular;
  cursor: pointer;
  margin-top: 5rem;
  :disabled {
    cursor: default;
    background-color: #a9afb8;
  }
`;
