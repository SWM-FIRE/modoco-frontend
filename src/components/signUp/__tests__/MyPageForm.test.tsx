import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import MyPageForm from '../MyPageForm';

describe('<MyPageForm />', () => {
  it('matches snapshot', () => {
    const utils = render(
      <Router>
        <MyPageForm />,
      </Router>,
    );
    expect(utils.container).toMatchSnapshot();
  });
  it('has a buttons', () => {
    const utils = render(
      <Router>
        <MyPageForm />,
      </Router>,
    );
    // MyPage에 있는지 확인
    utils.getByText('아바타 재생성');
    utils.getByText('프로필 이미지');
    utils.getByText('프로필 수정');
    utils.getByText('수정 완료');
  });
});
