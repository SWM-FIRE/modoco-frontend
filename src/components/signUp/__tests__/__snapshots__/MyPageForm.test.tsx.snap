// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyPageForm /> matches snapshot 1`] = `
<div>
  <div
    class="sc-ftvSup fRxRiZ"
  >
    <h1
      class="sc-papXJ cSNFdj"
    >
      프로필 수정
    </h1>
    <form
      class="sc-jqUVSM kPTbMi"
    >
      <div
        class="sc-bczRLJ fqGdDf"
      >
        <label
          class="sc-dkzDqf Revln"
        >
          프로필 이미지
        </label>
        <div
          class="sc-gsnTZi ddTkGA"
        >
          <svg
            fill="none"
            height="80"
            role="img"
            viewBox="0 0 36 36"
            width="80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Mahalia Jackson
            </title>
            <mask
              height="36"
              id="mask__beam"
              maskUnits="userSpaceOnUse"
              width="36"
              x="0"
              y="0"
            >
              <rect
                fill="#FFFFFF"
                height="36"
                rx="72"
                width="36"
              />
            </mask>
            <g
              mask="url(#mask__beam)"
            >
              <rect
                fill="#ffad08"
                height="36"
                width="36"
              />
              <rect
                fill="#73b06f"
                height="36"
                rx="36"
                transform="translate(2 6) rotate(222 18 18) scale(1)"
                width="36"
                x="0"
                y="0"
              />
              <g
                transform="translate(-6 1) rotate(-2 18 18)"
              >
                <path
                  d="M13,19 a1,0.75 0 0,0 10,0"
                  fill="#000000"
                />
                <rect
                  fill="#000000"
                  height="2"
                  rx="1"
                  stroke="none"
                  width="1.5"
                  x="12"
                  y="14"
                />
                <rect
                  fill="#000000"
                  height="2"
                  rx="1"
                  stroke="none"
                  width="1.5"
                  x="22"
                  y="14"
                />
              </g>
            </g>
          </svg>
          <button
            class="sc-hKMtZM fPBanq"
            type="button"
          >
            아바타 재생성
          </button>
        </div>
      </div>
      <p
        class="sc-eCYdqJ llvhwo"
      >
        <label
          class="sc-jSMfEi cbffqr"
          for="nickname"
        >
          닉네임 *
        </label>
        <input
          class="sc-gKXOVf iMMMEw"
          id="nickname"
          name="nickname"
          placeholder="닉네임을 입력해주세요."
          type="text"
          value=""
        />
        <span
          class="sc-iBkjds bBTOoc"
        />
      </p>
      <button
        class="sc-kDDrLX jgGKeV"
        disabled=""
      >
        수정 완료
      </button>
    </form>
  </div>
  ,
</div>
`;
